torchdata/dataloader2/adapter.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import functools
+
+from typing import Callable, Optional
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+
+__all__ = [
+    "Adapter",
+    "shuffle",
+]
+
+assert __all__ == sorted(__all__)
+
+Adapter = Callable[[IterDataPipe], IterDataPipe]
```

Comment:
Do you think it's better to provide a base class of callable `Adapter` ?
```py
class Adapter:
    @abstractmethod
    def __call__(self, datapipe: DataPipe) -> DataPipe:
        pass
```
- It's easier to check `isinstance(adapter, Adapter)`
- The type checking might be easier

Reply:
Surely it is. I just blanked out about this option. Thanks.